common/src/main/java/org/apache/drill/exec/metrics/CpuGaugeSet.java

```diff
@@ -32,13 +32,20 @@
  */
 @SuppressWarnings("restriction")
 public class CpuGaugeSet implements MetricSet {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CpuGaugeSet.class);
 
-  private OperatingSystemMXBean osMXBean;
-  private RuntimeMXBean rtMXBean;
+  private final OperatingSystemMXBean osMXBean;
+  private final RuntimeMXBean rtMXBean;
 
   public CpuGaugeSet() {
-    this.osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
     this.rtMXBean = ManagementFactory.getRuntimeMXBean();
+    //DRILL-6702: Instead of worrying about compiling with IBM JDK, for now, we shall provide no CPU metrics for non-HotSpot JVMs
+    if (System.getProperty("java.vm.name").contains("HotSpot")) {
```

Comment:
1.  We can extract `System.getProperty("java.vm.name")` into variable and not call it twice.
2. Can `System.getProperty("java.vm.name")` not be set and return null? In this case we'll get NPE.

Reply:
Where the guarantee that `ManagementFactory.getOperatingSystemMXBean()` returns an instance of `OperatingSystemMXBean` if `""java.vm.name"` contains HotSpot comes from? Why not to try the assignment and handle an exception?

Reply:
I'd recommend `try/catch` over `instanceOf`.

Reply:
Any particular reason why?

Reply:
- It is a common design pattern in the case when a check will result in a positive outcome 99.9% of the time.
- it can handle a JDK that decides not to implement `ManagementFactory.getOperatingSystemMXBean()` and throw `UnsupportedOperationException` instead

Reply:
@kkhatua The reverse of the condition is also problematic, it is possible that `"java.vm.name"` does not contain HotSpot while cast to `OperatingSystemMXBean` may still succeed.

Reply:
I like the `try/catch` approach. The instanceOf actually passed for the IBM JDK and failed in the cast. The `java.vm.name` appears to be standard, so I doubt it would be missing. Other JDKs might not implement it because it is a `com.sun. ...` package. (Remembering in context of Oracle vs Google, over Android)

Reply:
Sounds reasonable, let then proceed with try / catch approach.

Reply:
@kkhatua try GraalVM, it is a successor for HotSpot.

Reply:
Ok. I'm not able to repro the ClassCastException with the IBM JDK running an Oracle compilation. 
The IBM JDK on my Linux box is using `com.ibm.lang.management.internal.UnixExtendedOperatingSystem` which implements the `com.sun.management.OperatingSystemMXBean` interface. The JIRA reports a class cast exception from `com.ibm.lang.management.ExtendedOperatingSystem`. So, it might be an issue with the platform's implementation. 
I'll make the change however, and see if the original JIRA filer can confirm the fix.